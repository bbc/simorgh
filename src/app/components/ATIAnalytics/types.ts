import { PageTypes, Platforms, Services } from '../../models/types/global';
import { RequestContextProps } from '../../contexts/RequestContext';
import { ServiceConfig } from '../../models/types/serviceConfig';

export interface AMPAnalyticsData {
  transport: {
    beacon: boolean;
    xhrpost: boolean;
    image: boolean;
  };
  requests: {
    base?: string;
    pageview: string;
  };
  triggers: { trackPageview: { on: string; request: string } };
}

export interface PageData {
  metadata?: {
    analyticsLabels?: {
      counterName?: string;
      contentId?: string;
      // eslint-disable-next-line camelcase
      nations_producer?: string;
      pageIdentifier?: string;
      pageTitle?: string;
      producer?: string;
    };
    atiAnalytics?: {
      producerId?: string;
      chapter?: string;
    };
    locators?: { optimoUrn?: string; curie?: string };
    passport?: {
      category?: { categoryId?: string; categoryName?: string };
      campaigns?: { campaignId?: string; campaignName?: string }[];
      language?: string;
    };
    tags?: {
      about?: {
        thingId?: string;
        thingLabel?: string;
        thingEnglishLabel?: string;
      }[];
    };
    id?: string;
    title?: string;
    language?: string;
    firstPublished?: number;
    lastPublished?: number;
    typ?: PageTypes;
  };
  promo?:
    | { headlines?: { seoHeadline?: string } }
    | { headlines?: { headline?: string } }
    | { subType: string; name: string; uri: string; id: string; type: string };
  id?: string;
  language?: string;
  pageIdentifier?: string;
  pageTitle?: string;
  firstRecordTimeStamp?: string;
  lastRecordTimeStamp?: string;
  contentType?: string;
  title?: string;
}

export interface ATIData {
  campaigns?: { campaignId?: string; campaignName?: string }[] | null;
  categoryName?: string | null;
  contentId?: string | null;
  contentType?: string;
  language?: string | null;
  ldpThingIds?: string | null;
  ldpThingLabels?: string | null;
  nationsProducer?: string | null;
  pageIdentifier?: string;
  pageTitle?: string | null;
  producerId?: string | null;
  producerName?: string | null;
  timePublished?: string | null;
  timeUpdated?: string | null;
}

export interface ATIDataWithContexts {
  requestContext: RequestContextProps;
  serviceContext: ServiceConfig;
  atiData: ATIData;
}

export interface ATIConfigurationDetailsProviders {
  requestContext: RequestContextProps;
  serviceContext: ServiceConfig;
  data?: PageData;
  atiData?: ATIData;
}

export interface ATIAnalyticsProps {
  baseUrl?: string;
  pageviewParams: string;
}

export interface ATIEventTrackingProps {
  campaignID?: string;
  componentName?: string;
  format?: string;
  pageIdentifier?: string;
  platform?: Platforms;
  producerId?: string;
  service?: Services;
  statsDestination?: string;
  type?: string;
  advertiserID?: string;
  url?: string;
}

export interface ATIPageTrackingProps {
  appName?: string;
  contentId?: string | null;
  contentType?: string | null;
  language?: string | null;
  ldpThingIds?: string | null;
  ldpThingLabels?: string | null;
  pageIdentifier?: string;
  pageTitle?: string | null;
  producerId?: string;
  libraryVersion?: string;
  platform?: Platforms;
  statsDestination?: string;
  timePublished?: string | null;
  timeUpdated?: string | null;
  origin?: string;
  previousPath?: string | null;
  categoryName?: string | null;
  campaigns?: { campaignId?: string; campaignName?: string }[] | null;
  nationsProducer?: string | null;
}

export interface ATIProps {
  data?: PageData;
  atiData?: ATIData;
}
