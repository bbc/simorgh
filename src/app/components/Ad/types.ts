import { PageTypes, Services, Direction } from '#app/models/types/global';

export type SlotType = 'leaderboard' | 'mpu';

export interface AdProps {
  slotType: SlotType;
  className?: string;
}

type SlotFunction = (slotType: string) => void;

interface Ads {
  registerSlot: SlotFunction;
  destroySlot: SlotFunction;
}

type PushFunction = (slotFunction: SlotFunction) => void;

interface Cmd {
  push: PushFunction;
}

export interface WindowWithDotCom {
  cmd: Cmd;
  ads: Ads;
}

export interface CanonicalAdBootstrapJsProps {
  adcampaign?: string | null;
}

export interface AdJsonDataProps {
  service: Services;
  slotType: SlotType;
  assetType: string;
}

export interface AmpAdSlotProps {
  service: Services;
  slotType: SlotType;
  pageType: PageTypes;
}

export interface AmpAdContentProps extends AmpAdSlotProps {
  dir: Direction;
  label: string;
}

export interface AmpAdProps extends AmpAdContentProps {
  ariaLabel: string;
}
