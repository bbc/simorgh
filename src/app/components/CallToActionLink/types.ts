import { EventTrackingMetadata } from '#app/models/types/eventTracking';

export type CallToActionLinkProps = {
  href?: string;
  className?: string;
  eventTrackingData?: EventTrackingMetadata;
  download?: boolean;
};
