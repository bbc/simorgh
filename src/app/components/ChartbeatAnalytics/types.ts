import { Services } from '../../models/types/global';
import { MetadataTaggings } from '../../models/types/metadata';

type ChartbeatConfig = {
  domain: string;
  sections: string;
  uid: number;
  title: string;
  virtualReferrer: string | null;
  idSync?: {
    // eslint-disable-next-line camelcase
    bbc_hid: string;
  };
};

export type CanonicalChartbeatConfig = ChartbeatConfig & {
  type?: string;
  useCanonical?: boolean;
};

export interface CanonicalChartbeatProps {
  chartbeatConfig: CanonicalChartbeatConfig;
  chartbeatSource?: string;
}

export type AmpChartbeatConfig = ChartbeatConfig & {
  contentType?: string;
};

export interface AmpChartbeatProps {
  chartbeatConfig: AmpChartbeatConfig;
}

export interface ChartbeatProps {
  sectionName?: string; // required for STY and MAP pages
  categoryName?: string; // required for STY and MAP pages
  mediaPageType?: 'Radio' | 'TV' | 'Podcasts'; // required for Live & On Demand Radio, Podcasts & On Demand TV pages
  title: string;
  taggings?: MetadataTaggings; // required for Media Article Pages
  contentType?: 'player-live' | 'player-episode'; // required for Live & On Demand Radio, Podcasts & On Demand TV pages
  producer?: Services | string; // required for Media Article, STY, MAP, Live & On Demand Radio, Podcasts & On Demand TV pages, if available
  chapter?: string; // required for Media Article, STY, MAP, Live & On Demand Radio, Podcasts & On Demand TV pages, if available
}
