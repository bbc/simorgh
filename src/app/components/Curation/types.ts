import { Summary } from '#app/models/types/curationData';
import { Services, Variants } from '#models/types/global';

export interface Promo extends Summary {
  mediaType?: 'audio' | 'video' | 'photogallery';
  duration?: string | number;
  lazy?: boolean;
  headingLevel?: number;
}

export interface CurationGridProps {
  promos: Promo[];
  headingLevel?: number;
}

export interface StorybookProps {
  service: Services;
  variant: Variants;
}
