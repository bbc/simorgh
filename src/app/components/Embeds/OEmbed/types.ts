/* eslint-disable camelcase */
export type OEmbedProps = {
  oembed: {
    version: string;
    provider_name: string;
    provider_url: string;
    html: string;
    url?: string;
    source?: string;
    width?: number;
    height?: number;
    type: string;
  };
};
