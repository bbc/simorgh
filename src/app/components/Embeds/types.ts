/* eslint-disable camelcase */
export type ampParams = {
  'amp-clickable': boolean;
  'amp-image-height': number;
  'amp-image-width': number;
  'amp-image': string;
};

export type OEmbedData = Partial<{
  version: string;
  provider_name: string;
  provider_url: string;
  html: string;
  url: string;
  source: string;
  width: number;
  height: number;
  type: string;
  iFrameSrc: string;
  iFrameId: string;
  iFrameTitle: string;
  sizeAdjustScript: string;
  parameters: ampParams;
  oEmbedType: string;
}>;

export type OEmbedProps = {
  type: string;
  oembed: OEmbedData;
};
