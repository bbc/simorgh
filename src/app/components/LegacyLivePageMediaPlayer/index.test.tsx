import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '../react-testing-library-with-providers';
import LivePageMediaPlayer from '.';
import { validLivePageVideoWithCaptionBlock } from '../../legacy/containers/MediaPlayer/fixtureData';
import { LIVE_PAGE } from '#routes/utils/pageTypes';
import { MediaBlock } from '../MediaLoader/types';

const GenerateMediaPlayer = ({ blocks }: { blocks: MediaBlock[] }) => (
  <BrowserRouter>
    <LivePageMediaPlayer blocks={blocks} className="MediaPlayer" />
  </BrowserRouter>
);

describe('LegacyLiveMediaPlayer', () => {
  it('Calls the canonical media player, with a placeholder', () => {
    const { container } = render(
      <GenerateMediaPlayer
        blocks={validLivePageVideoWithCaptionBlock as MediaBlock[]}
      />,
      {
        id: 'c7p765ynk9qt',
        service: 'pidgin',
        pageType: LIVE_PAGE,
        pathname: '/pidgin/live/c7p765ynk9qt',
        statusCode: 200,
      },
    );
    expect(container).toMatchSnapshot();
  });

  it('Calls the amp media player', () => {
    const { container } = render(
      <GenerateMediaPlayer
        blocks={validLivePageVideoWithCaptionBlock as MediaBlock[]}
      />,
      {
        isAmp: true,
        id: 'c7p765ynk9qt',
        service: 'pidgin',
        pageType: LIVE_PAGE,
        pathname: '/pidgin/live/c7p765ynk9qt',
        statusCode: 200,
      },
    );
    expect(container).toMatchSnapshot();
  });
});
