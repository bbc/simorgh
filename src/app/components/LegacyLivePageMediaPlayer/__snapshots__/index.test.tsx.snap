// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegacyLiveMediaPlayer Calls the amp media player 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }
}

.emotion-2 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-4 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-6 {
  position: relative;
  height: 0;
  overflow: hidden;
  background-color: #F2F2F2;
  -webkit-background-position: center center;
  background-position: center center;
  background-repeat: no-repeat;
  -webkit-background-size: 60px 17px;
  background-size: 60px 17px;
  width: 100%;
  background-image: url(data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz48c3ZnIHdpZHRoPSI0NzkiIGhlaWdodD0iMTM2IiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciPjxnIGZpbGw9Im5vbmUiIGZpbGwtcnVsZT0iZXZlbm9kZCI+PGcgZmlsbD0iI0Q1RDBDRCI+PGc+PHBhdGggZD0iTTQ0OS41NTYgOTEuNTUzbC0yLjMxIDEuNDFjLTExLjE2NyA2LjgxOC0yMy4zMTMgMTAuNDc0LTM0LjM0NiAxMC40Ny0yMy42MS0uMDktMzkuMTYzLTE0LjA4My0zOS4yMjctMzQuNDUzLjAzLTE5LjkxOCAxNi4yNy0zNC42NjMgMzguNjMzLTM0LjcxOCAxMS4zODcuMDEgMjEuNzAzIDIuOTU0IDMzLjk2MiA5LjY3MmwyLjI1MSAxLjI0di0xOC4xOWwtLjk2Mi0uMzc3Yy0xMy44MjQtNS40NTgtMjQuNTMtNy44OS0zNS4xMDMtNy44ODQtMTYuMzQ2LS4wMDYtMzAuNTMzIDUuMzk0LTQwLjYzNyAxNC41NTctMTAuMTA1IDkuMTYzLTE2LjEwNiAyMi4xMDItMTYuMDk5IDM2Ljk1My4wMDggMTAuMzQ4IDQuMjc5IDIyLjQ4IDEzLjQyIDMyLjEwNSA5LjEyMSA5LjYyOCAyMy4xNjUgMTYuNjQ4IDQyLjQzIDE2LjYzOWguMDYzYzE1Ljk4IDAgMjcuMDYyLTMuNTYzIDM3LjA3NC04LjQ5MmwuODUxLS40MTRWOTEuNTUzek0zMzQgMTM1LjY5N2gxNDQuMTk1VjBIMzM0djEzNS42OTd6Ii8+PHBhdGggZD0iTTI3Ni45MzcgODkuOTY4Yy4wNDEtMTIuMzMtOC4xNzEtMjEuNjk2LTIxLjMwOC0yNS4zIDMuNTQ0LTEuODA5IDYuMzUtNC4wMjMgOC40MDQtNi43MjcgMi43NS0zLjYyMiA0LjA2MS04LjA2NCA0LjA0Ni0xMy4yMzUuMDE1LTYuMzU5LTIuNDg2LTEyLjgzOS03Ljg1OC0xNy42ODctNS4zNzItNC44NDctMTMuNTI2LTcuOTk3LTI0LjY1NC03Ljk5MUgyMDQuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMTY3IDEzNS42OThoMTQ0LjE5N1YwSDE2N3YxMzUuNjk3eiIvPjxwYXRoIGQ9Ik0xMDkuOTM3IDg5Ljk2OGMuMDQxLTEyLjMzLTguMTcxLTIxLjY5Ni0yMS4zMDgtMjUuMyAzLjU0NC0xLjgwOSA2LjM1LTQuMDIzIDguNDA0LTYuNzI3IDIuNzUtMy42MjIgNC4wNjEtOC4wNjQgNC4wNDYtMTMuMjM1LjAxNS02LjM1OS0yLjQ4Ni0xMi44MzktNy44NTgtMTcuNjg3LTUuMzcyLTQuODQ3LTEzLjUyNi03Ljk5Ny0yNC42NTQtNy45OTFIMzcuODN2OTcuNzI4aDM2LjA3M2MxMi44NyAwIDIxLjkwNi0zLjQ4MiAyNy43MjItOC42NSA1LjgxOC01LjE1NSA4LjMyLTExLjkxIDguMzEyLTE4LjEzOHpNMCAxMzUuNjk4aDE0NC4xOTdWMEgwdjEzNS42OTd6Ii8+PHBhdGggZD0iTTI1OC42NjIgODguMTk4Yy0uMDEzIDMuMjI5LTEuMDA3IDYuNDc1LTMuODk2IDkuMDExLTIuODg0IDIuNTM3LTcuODczIDQuNDYzLTE2LjEzMyA0LjQ2M0gyMjJWNzVoMTUuODkzYzcuNDExIDAgMTIuNjcgMS41MDIgMTUuOTY1IDMuODUgMy4yODkgMi4zNjIgNC43NzYgNS40NjMgNC44MDQgOS4zNDgiLz48cGF0aCBkPSJNOTEuNjYyIDg4LjE5OGMtLjAxMyAzLjIyOS0xLjAwNyA2LjQ3NS0zLjg5NiA5LjAxMS0yLjg4NCAyLjUzNy03Ljg3NCA0LjQ2My0xNi4xMzMgNC40NjNINTVWNzVoMTUuODkyYzcuNDEyIDAgMTIuNjcyIDEuNTAyIDE1Ljk2NiAzLjg1IDMuMjg5IDIuMzYyIDQuNzc2IDUuNDYzIDQuODA0IDkuMzQ4Ii8+PHBhdGggZD0iTTI0NS4xODYgNTUuNzljMy4wOTYtMi4yMzcgNC41OS01LjM4NiA0LjYxMy0xMC4xMjQtLjAxNS0zLjI1LS45NDMtNi4wMzMtMy4yODEtOC4xMTEtMi4zNDYtMi4wNzgtNi4zMy0zLjU1NS0xMi43NTQtMy41NTVIMjIydjI1LjI3NWg4LjA3NmM2Ljk4OC4wMDQgMTEuOTk4LTEuMjQzIDE1LjExLTMuNDg2Ii8+PHBhdGggZD0iTTc4LjE4NiA1NS43OWMzLjA5Ni0yLjIzNyA0LjU5LTUuMzg2IDQuNjEzLTEwLjEyNC0uMDE1LTMuMjUtLjk0My02LjAzMy0zLjI4Mi04LjExMUM3Ny4xNzIgMzUuNDc3IDczLjE4OCAzNCA2Ni43NjQgMzRINTV2MjUuMjc1aDguMDc2YzYuOTg4LjAwNCAxMS45OTgtMS4yNDMgMTUuMTEtMy40ODYiLz48L2c+PC9nPjwvZz48L3N2Zz4K);
}

@media (min-width: 25rem) {
  .emotion-6 {
    -webkit-background-size: 77px 22px;
    background-size: 77px 22px;
  }
}

@media (min-width: 63rem) {
  .emotion-6 {
    -webkit-background-size: 93px 27px;
    background-size: 93px 27px;
  }
}

.emotion-8 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #545658;
  margin-top: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;
  width: calc(100% - 0.5rem);
  margin-left: 0.5rem;
  border-left: 0.0625rem solid #6E6E73;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-8 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-8 {
    width: 100%;
    margin: 0.5rem 0 0;
  }
}

.emotion-8>span>p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.emotion-8>span>p:last-child {
  padding-bottom: 0;
}

@media (min-width: 25rem) and (max-width: 62.9375rem) {
  .emotion-8 {
    width: calc(100% - 1rem);
    margin-left: 1rem;
    padding-right: 1rem;
    padding-left: 0.5rem;
  }
}

@media (min-width: 63rem) {
  .emotion-8 {
    padding-right: 0;
    padding-left: 0.5rem;
  }
}

.emotion-10 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

<div>
  <div
    class="MediaPlayer emotion-0 emotion-1"
    dir="ltr"
  >
    <figure
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <amp-iframe
          allowfullscreen="allowfullscreen"
          data-e2e="media-player"
          frameborder="0"
          layout="fill"
          sandbox="allow-scripts allow-same-origin"
          scrolling="no"
          src="https://web-cdn.test.api.bbci.co.uk/ws/av-embeds/live/c7p765ynk9qt/p01thw20/pcm/amp"
          title="Media player"
        >
          <div
            class="emotion-6 emotion-7"
            data-e2e="image-placeholder"
            placeholder=""
            style="padding-bottom: 56.25%;"
          />
          <noscript />
        </amp-iframe>
      </div>
      <figcaption
        class="emotion-8 emotion-9"
        dir="ltr"
      >
        <span
          role="text"
        >
          <span
            class="emotion-10"
          >
            Wetin we call dis Video, 
          </span>
          <span
            data-testid="caption-paragraph"
          >
            BBC launch trailer for We Know Our Place women's sport campaign"
          </span>
        </span>
      </figcaption>
    </figure>
  </div>
</div>
`;

exports[`LegacyLiveMediaPlayer Calls the canonical media player, with a placeholder 1`] = `
@media (max-width: 14.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 15rem) and (max-width: 24.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 25rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 37.5rem) and (max-width: 62.9375rem) {
  .emotion-0 {
    margin-left: 0%;
  }
}

@media (min-width: 63rem) and (max-width: 79.9375rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@media (min-width: 80rem) {
  .emotion-0 {
    margin-left: 20%;
  }
}

@supports (display: grid) {
  .emotion-0 {
    display: block;
    width: initial;
    margin: 0;
  }

  @media (max-width: 14.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 15rem) and (max-width: 24.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 25rem) and (max-width: 37.4375rem) {
    .emotion-0 {
      grid-template-columns: repeat(6, 1fr);
      grid-column-end: span 6;
      grid-column-start: 1;
    }
  }

  @media (min-width: 37.5rem) and (max-width: 62.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 1;
    }
  }

  @media (min-width: 63rem) and (max-width: 79.9375rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }

  @media (min-width: 80rem) {
    .emotion-0 {
      grid-template-columns: repeat(5, 1fr);
      grid-column-end: span 5;
      grid-column-start: 2;
    }
  }
}

.emotion-2 {
  margin: 0;
  padding-bottom: 1.5rem;
  width: 100%;
}

.emotion-4 {
  padding-top: 56.25%;
  position: relative;
  overflow: hidden;
}

.emotion-6 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  cursor: default;
}

.emotion-6:hover .emotion-12,
.emotion-6:focus .emotion-12 {
  background-color: #B80000;
}

.emotion-8 {
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.9375rem;
  line-height: 1.125rem;
  width: 100%;
  height: 100%;
  position: absolute;
  border: 0.0625rem solid transparent;
  color: #FFFFFF;
}

@media (min-width: 37.5rem) {
  .emotion-8 {
    font-size: 0.875rem;
  }
}

.no-js .emotion-8 {
  background-color: rgba(34, 34, 34, 0.75);
}

@media screen and (-ms-high-contrast: active) {
  .no-js .emotion-8 {
    background-color: transparent;
  }
}

.no-js .emotion-8 .guidance-message {
  display: none;
}

.emotion-10 {
  position: absolute;
  bottom: 0;
}

.emotion-13 {
  background-color: #222222;
  border: none;
  color: #FFFFFF;
  cursor: pointer;
  display: block;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 0.75rem;
  line-height: 1rem;
  height: 5rem;
  padding: 0;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
  width: 5rem;
  position: absolute;
  bottom: 0;
  left: 0;
}

.emotion-13:hover,
.emotion-13:focus {
  background-color: #B80000;
  -webkit-transition: background-color 300ms;
  transition: background-color 300ms;
}

.no-js .emotion-13 {
  display: none;
}

.emotion-13:focus-visible::before {
  content: '';
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  box-shadow: 0 0 0 0.1875rem #FFFFFF inset;
  border: 0.1875rem solid #000000;
}

.emotion-15 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  clip: rect(1px, 1px, 1px, 1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
  margin: 0;
}

.emotion-16>svg {
  color: #FFFFFF;
  fill: currentColor;
  height: 1.5rem;
  margin-top: 0.5rem;
  width: 1.5rem;
}

.emotion-18 {
  vertical-align: middle;
  margin: 0 0.25rem;
  color: #222222;
  fill: currentColor;
  width: 0.75rem;
  height: 0.75rem;
}

.emotion-20 {
  display: block;
  margin-top: 0.5rem;
}

.emotion-22 {
  display: block;
  width: 100%;
}

.emotion-24 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  font-family: Helmet,Freesans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  color: #545658;
  margin-top: 0.5rem;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  width: 100%;
  width: calc(100% - 0.5rem);
  margin-left: 0.5rem;
  border-left: 0.0625rem solid #6E6E73;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-24 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-24 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 63rem) {
  .emotion-24 {
    width: 100%;
    margin: 0.5rem 0 0;
  }
}

.emotion-24>span>p {
  padding-bottom: 1.5rem;
  margin: 0;
}

.emotion-24>span>p:last-child {
  padding-bottom: 0;
}

@media (min-width: 25rem) and (max-width: 62.9375rem) {
  .emotion-24 {
    width: calc(100% - 1rem);
    margin-left: 1rem;
    padding-right: 1rem;
    padding-left: 0.5rem;
  }
}

@media (min-width: 63rem) {
  .emotion-24 {
    padding-right: 0;
    padding-left: 0.5rem;
  }
}

<div>
  <div
    class="MediaPlayer emotion-0 emotion-1"
    dir="ltr"
  >
    <figure
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
        data-e2e="media-player"
      >
        <div
          class="emotion-6 emotion-7"
          data-e2e="media-player__placeholder"
        >
          <div
            class="emotion-8 emotion-9"
            data-e2e="media-player__guidance"
          >
            <noscript
              class="emotion-10 emotion-11"
            />
          </div>
          <button
            class="focusIndicatorRemove emotion-12 emotion-13 emotion-14"
          >
            <span
              class="emotion-15"
            >
              Play video, "BBC launch trailer for We Know Our Place women's sport campaign", Duration 0,54
            </span>
            <div
              aria-hidden="true"
              class="emotion-16 emotion-17"
            >
              <svg
                aria-hidden="true"
                class="emotion-18 emotion-19"
                focusable="false"
                height="12"
                viewBox="0 0 12 12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M.5.6h12v12H.5z"
                  />
                  <path
                    d="M2.144.96v11.28l8.712-5.64z"
                    fill="currentColor"
                  />
                </g>
              </svg>
            </div>
            <time
              aria-hidden="true"
              class="emotion-20 emotion-21"
              datetime="PT54S"
            >
              00:54
            </time>
          </button>
          <img
            alt=""
            class="emotion-22 emotion-23"
            src="https://ichef.test.bbci.co.uk/images/ic/512xn/p01thw3g.jpg.webp"
          />
        </div>
      </div>
      <figcaption
        class="emotion-24 emotion-25"
        dir="ltr"
      >
        <span
          role="text"
        >
          <span
            class="emotion-15"
          >
            Wetin we call dis Video, 
          </span>
          <span
            data-testid="caption-paragraph"
          >
            BBC launch trailer for We Know Our Place women's sport campaign"
          </span>
        </span>
      </figcaption>
    </figure>
  </div>
</div>
`;
